app/scripts/modules/core/src/utils/failedToInstantiateModule.ts

```diff
@@ -0,0 +1,13 @@
+import * as angular from 'angular';
+import { IModule } from 'angular';
+
+// Improve errors/stack traces related to:
+// `Error: [ng:areq] Argument 'module' is not a function, got undefined`
+const realModule = angular.module;
+(angular as any).module = function(name: string, requires?: string[], configFn?: Function): IModule {
+  if (requires && requires.some(dep => !dep)) {
+    const deps = requires.map(r => typeof r === 'string' ? `'${r}'` : `${r}`).join(', ');
+    throw new Error(`Got falsey dependency whilst registering angular module '${name}': [${deps}]`);
```

Comment:
do we care about whether or not `angular.module` gets deoptimized by V8? `throw` inside a function is an auto-deopt. My guess is it's not a big deal since it likely only runs a few hundred times during load before the optimization path even has a good chance to kick in.

Reply:
I don't care. can't speak for the other nerds

Reply:
Too nerdy for me.

Reply:
I'm curious if it makes a difference. There's a lot of time spent loading JS when the page first starts up but I am _guessing_ this won't add enough to it to be noticeable. We can do some checks once it's landed in a production environment and roll back if it does.

Reply:
I'm wagering this won't be noticeable much or maybe at all, but wasn't sure if we'd previously seen a bottleneck on module resolution.

Reply:
I’ll take my four months if the difference is even measurable